model/generate.go

```diff
@@ -1,25 +1,296 @@
 package model
 
 import (
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/model/version"
+	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
 	yaml "gopkg.in/yaml.v2"
 )
 
-// GeneratedTasks is a subset of the Project type, and is generated from the
+const (
+	maxGeneratedBuildVariants = 100
+	maxGeneratedTasks         = 1000
+)
+
+// GeneratedProjects is a convenience type for a slice of generated projects.
+type GeneratedProjects []GeneratedProject
```

Comment:
the difference of just an S, I think hinders readability. I'd avoid doing this unless you need to define methods on the slice.

Reply:
done